### ISSUE TITLE
Small issue

### ISSUE THREAD

**Issue by 911reg** [EliotVU/Unreal-Library#56]
Hey there Eliot, first of all, thanks again for the great app! This issue isn't really related to a bug or a problem with the app, it's more like a question, if solved it might be useful for someone else as well.

So basically, i've been trying to extract *.uc files from a file ([this](https://cdn.discordapp.com/attachments/725065259806883890/1061201969487810590/LineageSkillEffect.u) one specifically)

The problem i'm having is the following: 
This is what the defaultproperties look like:

![Sin título](https://user-images.githubusercontent.com/70298452/211142385-1cbf13a2-d2b3-414e-96b0-78159535867e.png)

when they should look like this: 

![Screenshot_2](https://user-images.githubusercontent.com/70298452/211142419-19727328-2849-440c-a883-53336b478d0a.jpg)

So basically it's showing up as "**_L2EffectEmitter_**" instead of "**_L2EffectEmiter'L2EffectEmiter0_**'

I'm 100% doing something wrong here, do you perhaps know how to solve this?
Here's the Array Definition i'm using: **_Package.Class.PreshotAction:ObjectProperty_**

Also, when i see the script inside UE Explorer, the reference is there, but it isn't applied to the ShotAction line for whatever reason

    // Reference: L2EffectEmitter'l2_14207_skill.L2EffectEmitter53'
    ShotAction(0)=L2EffectEmitter53
![Screenshot_1](https://user-images.githubusercontent.com/70298452/211142320-7bb11758-c862-4112-9c3a-5c483386af1f.jpg)

PD: i can fix this by hand, but sadly, i'd have to do that manually for over 4000 files.

**Comment by EliotVU** [comment]
Are you trying to re-compile the code? If so, either references should be fine, the compiler generally just looks up the object by the name's hash.

Nonetheless I'll label this issue as an enhancement request to output a qualified reference instead.

**Comment by 911reg** [comment]
I'm trying to recompile, yeah, but i'm getting the 'unknown property' warning in ucc.exe. The file gets compiled, but ot doesn't work ingame, though it works if i add the part that's missing manually

![Screenshot_3](https://user-images.githubusercontent.com/70298452/211167848-88699196-363d-4699-a211-52de7a251a92.jpg)

**Comment by EliotVU** [comment]
Strange, normally the compiler understands such references, maybe it's different for L2.

I will address this issue in the next release by always outputting a fully qualified reference.

**Comment by EliotVU** [comment]
Related issue #40

**Comment by 911reg** [comment]
Nice! I just ended up making a small tool to "fix" the files for me, it worked as a temporary measure xD
i'll leave it here just in case

```
#include <iostream>
#include <fstream>
#include <dirent.h>
#include <string>

using namespace std;

int main() {
    DIR* dir;
    struct dirent* ent;
    if ((dir = opendir(".")) != NULL) {
        while ((ent = readdir(dir)) != NULL) {
            string fileName = ent->d_name;
            if (fileName.find(".uc") != string::npos) {
                // found a .uc file
                cout << "Editing file: " << fileName << endl;
                ifstream inputFile(fileName);
                string line;
                string editedFile;
                while (getline(inputFile, line)) {
                    if (line.find("Action(") != string::npos) {
                        // found a line with "Action("
                        int equalPos = line.find("=");
                        if (equalPos != string::npos) {
                            // found the "=" symbol on the same line as "Action("
                            line.replace(equalPos, 1, "=L2EffectEmitter'");
                            line += "'";
                        }
                    }
                    editedFile += line + "\n";
                }
                inputFile.close();
                // overwrite the .uc file with the edited content
                ofstream outputFile(fileName);
                outputFile << editedFile;
                outputFile.close();
            }
        }
        closedir(dir);
    }
    else {
        cout << "Error opening directory" << endl;
    }
    return 0;
}
```
